src/main/java/edu/nicholaidudakiwwarrick/advancedjava/services/DatabasePersonService.java

```diff
@@ -0,0 +1,155 @@
+package edu.nicholaidudakiwwarrick.advancedjava.services;
+
+import edu.nicholaidudakiwwarrick.advancedjava.model.DatabasePerson;
+import edu.nicholaidudakiwwarrick.advancedjava.model.DatabasePersonStock;
+import edu.nicholaidudakiwwarrick.advancedjava.model.DatabaseStockSymbol;
+import edu.nicholaidudakiwwarrick.advancedjava.util.DatabaseUtils;
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.criterion.Restrictions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is a concrete implementation of PersonService that reads data from a database.
+ *
+ * @author Nicholai Dudakiw-Warrick
+ */
+public class DatabasePersonService implements PersonService {
+
+    public static final DatabasePersonService INSTANCE = new DatabasePersonService();
+
+    // hides the constructor and delegates instance creation through ServiceFactory
+    protected DatabasePersonService() {
+    }
+
+    /**
+     * Gets a list of all persons stored in the "person" database.
+     * Changes made by mutating objects in the returned list do not propagate to the database.
+     * @return a list of Person instances
+     * @throws PersonServiceException if a service can not perform the requested operation
+     */
+    public final List<DatabasePerson> getPersons() throws PersonServiceException {
+        Session session = DatabaseUtils.getSessionFactory().openSession();
+        List<DatabasePerson> persons = null;
+        Transaction transaction = null;
+        try {
+            // assigns all instances of Person within session to Criteria
+            // then converts Criteria to a list of the criteria type which is Person.class
+            transaction = session.beginTransaction();
+            Criteria criteria = session.createCriteria(DatabasePerson.class);
+
+            // intentional suppression of warnings for unchecked exceptions thrown by criteria.list()
+            @SuppressWarnings("unchecked") List<DatabasePerson> list = criteria.list();
+            persons = list;
+        } catch (HibernateException e) {
+            if (transaction != null && transaction.isActive()) {
+                transaction.rollback();  // close transaction
+            }
+            throw new PersonServiceException("Could not get Person data. " + e.getMessage(), e);
```

Comment:
so this might be a problem because the finally statement will always get called.